src/components/loginForm/LoginForm.tsx

```diff
@@ -1,55 +1,106 @@
-import React, { useState } from 'react';
+import React, { useState, useContext, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 import styles from './LoginForm.module.css';
+import { HttpStatusCode } from '../../http-codes';
+import AuthContext from '../../AuthContext';
 
-interface LoginFormProps {
-  onSubmit: (username: string, password: string) => void;
-}
-
-function LoginForm({ onSubmit }: LoginFormProps) {
-  const [username, setUsername] = useState('');
+function LoginForm() {
+  const [inputUsername, setInputUsername] = useState('');
   const [password, setPassword] = useState('');
+  const [error, setError] = useState<string | null>(null);
+  const navigate = useNavigate();
+  const { setAuthData, username } = useContext(AuthContext);
+
+  const APIBASE = 'http://localhost:3000/api/';
 
-  const handleUsernameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    setUsername(e.target.value);
+  useEffect(() => {
+    if (username) {
+      navigate('/');
+    }
+  }, []);
+
+  const handleinputUsernameChange = (
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setInputUsername(e.target.value);
   };
 
   const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     setPassword(e.target.value);
   };
 
-  const handleSubmit = (e: React.FormEvent) => {
-    e.preventDefault();
-    onSubmit(username, password);
+  const handleSubmit = async () => {
```

Comment:
`handleSubmit` needs to prevent the default behavior of the html form

Add *e* as an argument to handleSubmit with type React.FormEvent<HTMLFormElement>

```typescript
  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => // ... 
```